sample_project/Assets/SampleViewer/Samples/HitTest/ArcGISRaycast.cs

```diff
@@ -52,39 +133,50 @@ private void OnLeftShift(bool isPressed)
 
     private void OnLeftClickStart(InputAction.CallbackContext context)
     {
-        if (isLeftShiftPressed)
+        if (isLeftShiftPressed && !EventSystem.current.IsPointerOverGameObject())
         {
-            if (!canvas.enabled)
-            {
-                canvas.enabled = true;
-            }
             RaycastHit hit;
             Ray ray = Camera.main.ScreenPointToRay(Mouse.current.position.ReadValue());
 
             if (Physics.Raycast(ray, out hit))
             {
                 var arcGISRaycastHit = arcGISMapComponent.GetArcGISRaycastHit(hit);
                 var layer = arcGISRaycastHit.layer;
-                var featureId = arcGISRaycastHit.featureId;
+                featureId = arcGISRaycastHit.featureId;
+                
+                CreateLink(featureId.ToString());
 
-                if (layer != null && featureId != -1)
+                if (layer != null)
                 {
-                    featureText.text = featureId.ToString();
-
                     var geoPosition = arcGISMapComponent.EngineToGeographic(hit.point);
-                    var offsetPosition = new ArcGISPoint(geoPosition.X, geoPosition.Y, geoPosition.Z + offSet, geoPosition.SpatialReference);
-
-                    var rotation = arcGISCamera.GetComponent<ArcGISLocationComponent>().Rotation;
-                    var location = canvas.GetComponent<ArcGISLocationComponent>();
-                    location.Position = offsetPosition;
-                    location.Rotation = rotation;
+                    var location = markerGO.GetComponent<ArcGISLocationComponent>();
+                    location.Position = new ArcGISPoint(geoPosition.X, geoPosition.Y, geoPosition.Z, geoPosition.SpatialReference);
+                    
+                    var point = ArcGISGeometryEngine.Project(geoPosition, ArcGISSpatialReference.WGS84()) as ArcGISPoint;
+                    var lat = Mathf.Round((float)point.Y * 1000) / 1000;
+                    var lon = Mathf.Round((float)point.X * 1000) / 1000; 
+                    locationText.text = "Lat: " + lat + " Long: " + lon;
```

Comment:
```suggestion
                    var point = ArcGISGeometryEngine.Project(geoPosition, ArcGISSpatialReference.WGS84()) as ArcGISPoint;
                    locationText.text = $"Lat: {string.Format("{0:0.##}", point.Y)} Long: {string.Format("{0:0.##}", point.X)}";
```

Changing number display can be done with `String.Format`